### ISSUE TITLE
Issue with ML.Dev.Prog.Sig Function: Infinite Argument in coxph.wtest

### ISSUE THREAD

**Issue by ZhenWang325429** [l-magnificence/Mime#24]
Dear,

I am currently utilizing the Mime1 package for my bioinformatics research and have encountered an issue with the ML.Dev.Prog.Sig function. Specifically, the following code results in an error:

Res <- ML.Dev.Prog.Sig(
  train_data = Train_Valition[[1]],
  list_train_vali_Data = Train_Valition,
  unicox.filter.for.candi = TRUE,
  unicox_p_cutoff = 0.05,
  candidate_genes = genelist,
  mode = 'all',
  nodesize = 5,
  seed = 5201314)

The error message returned is:
Error in coxph.wtest(fit$var[nabeta, nabeta], temp, control$toler.chol) : 
  infinite argument in coxph.wtest
Our Train_Valition list contains three elements, where the first element is the training set and the remaining two elements are validation sets.

I have verified that the data does not contain any obvious anomalies such as infinite or missing values. Could you provide guidance on what might be causing this error and how to resolve it? Any assistance or suggestions would be greatly appreciated.

Thank you for your help and for developing such a valuable tool.

Best regards,

**Comment by l-magnificence** [comment]
Hi, can you show your detailed log in which step the error occurred?

**Comment by zifeng17** [comment]
load("./Example.cohort.Rdata")
list_train_vali_Data[["Dataset1"]][1:5,1:5]
load("./Example.ici.Rdata")
list_train_vali_Data[["training"]][1:5,1:5]
load("./genelist.Rdata")
library(Mime1)
load("./Example.cohort.Rdata")
load("./genelist.Rdata")
res <- ML.Dev.Prog.Sig(train_data = list_train_vali_Data$Dataset1,
                       list_train_vali_Data = list_train_vali_Data,
                       unicox.filter.for.candi = T,
                       unicox_p_cutoff = 0.1,
                       candidate_genes = genelist,
                       mode = 'all',nodesize =5,seed = 5201314 )


look at this error：
                    
--- 10.Lasso + superPC ---
Error: infinite argument in coxph.wtest 
Retrying...
--- 10.Lasso + survival-SVM ---
There were 50 or more warnings (use warnings() to see the first 50)
> list_train_vali_Data.head(5)
Error in list_train_vali_Data.head(5) : 
  could not find function "list_train_vali_Data.head"

**Comment by l-magnificence** [comment]
> load("./Example.cohort.Rdata") list_train_vali_Data[["Dataset1"]][1:5,1:5] load("./Example.ici.Rdata") list_train_vali_Data[["training"]][1:5,1:5] load("./genelist.Rdata") library(Mime1) load("./Example.cohort.Rdata") load("./genelist.Rdata") res <- ML.Dev.Prog.Sig(train_data = list_train_vali_Data$Dataset1, list_train_vali_Data = list_train_vali_Data, unicox.filter.for.candi = T, unicox_p_cutoff = 0.1, candidate_genes = genelist, mode = 'all',nodesize =5,seed = 5201314 )
> 
> look at this error：
> 
> --- 10.Lasso + superPC --- Error: infinite argument in coxph.wtest Retrying... --- 10.Lasso + survival-SVM --- There were 50 or more warnings (use warnings() to see the first 50)
> 
> > list_train_vali_Data.head(5)
> > Error in list_train_vali_Data.head(5) :
> > could not find function "list_train_vali_Data.head"

It run successfully. Ignore it. When Mime encounter an error in this method, it have retried.
list_train_vali_Data.head is not our function in Mime.

**Comment by zifeng17** [comment]
感谢您的回复！


但是重点在这里：|
--- 10.Lasso + StepCox[forward] ---
Start:  AIC=178.09
Surv(OS.time, OS) ~ CTNNB1 + DLL1 + AXIN2 + PSEN2 + WNT5B + DKK1 + 
    NCSTN + SKP2 + HEY1

--- 10.Lasso + superPC ---
Error: infinite argument in coxph.wtest 
Retrying...
--- 10.Lasso + survival-SVM ---
There were 50 or more warnings (use warnings() to see the first 50)

**Comment by l-magnificence** [comment]
> 感谢您的回复！
> 
> 但是重点在这里：| --- 10.Lasso + StepCox[forward] --- Start: AIC=178.09 Surv(OS.time, OS) ~ CTNNB1 + DLL1 + AXIN2 + PSEN2 + WNT5B + DKK1 + NCSTN + SKP2 + HEY1
> 
> --- 10.Lasso + superPC --- Error: infinite argument in coxph.wtest Retrying... --- 10.Lasso + survival-SVM --- There were 50 or more warnings (use warnings() to see the first 50)

Sorry, I don't know what you mean? There is no error.

**Comment by zifeng17** [comment]
The error message returned is:
Error in coxph.wtest(fit$var[nabeta, nabeta], temp, control$toler.chol) :
infinite argument in coxph.wtest

**Comment by l-magnificence** [comment]
see #40

**Comment by wjlc1992** [comment]
> load("./Example.cohort.Rdata") list_train_vali_Data[["Dataset1"]][1:5,1:5] load("./Example.ici.Rdata") list_train_vali_Data[["training"]][1:5,1:5] load("./genelist.Rdata") library(Mime1) load("./Example.cohort.Rdata") load("./genelist.Rdata") res <- ML.Dev.Prog.Sig(train_data = list_train_vali_Data$Dataset1, list_train_vali_Data = list_train_vali_Data, unicox.filter.for.candi = T, unicox_p_cutoff = 0.1, candidate_genes = genelist, mode = 'all',nodesize =5,seed = 5201314 )
> 
> look at this error：
> 
> --- 10.Lasso + superPC --- Error: infinite argument in coxph.wtest Retrying... --- 10.Lasso + survival-SVM --- There were 50 or more warnings (use warnings() to see the first 50)
> 
> > list_train_vali_Data.head(5)
> > Error in list_train_vali_Data.head(5) :
> > could not find function "list_train_vali_Data.head"

Hi, I encountered the same problem running Mime1 package, could I ask that have you solved your problem? Thanks
